### DISCUSSION TITLE
Can I render React Native components on the web?

### DISCUSSION THREAD

**Discussion by just-Bri** [necolas/react-native-web#2088] [Q&A] [answered]
I've been trying to google this for weeks and it's been extremely hard to actually find the question/answer I'm looking for.  

Is there any way for RNW to take a ReactNative component, for example:
```typescript
const UserName: React.FC<{ user: userObject}> =({ user} => {
  return (
    <View><Text>{user.name}</Text></View>
  );
}
```  
And give me a react usable component?

The app I'm working on is very large and converting the entire app to RNW isn't currently an option, but we also have a RN app for iOS. My goal is to be able to create a RN component in a shared library and use the base RN component on mobile, and use a generated/compiled version in our React web app.

If anyone has any insight of can point me in the right direction please let me know.  
Or this is simply not possible given the current capabilities of React/RN/RNW please tell me so I can put the shovel down!

**Comment by necolas** [comment, marked as answer]
You can run this code with RNW alongside existing React DOM code

**Reply by just-Bri** [reply]
~~Do you mean I would be able to just import and call/render that component in my web app using RNW?  
If that's the case is it as simple as adding/changing some webpack configs to support RN?~~

I feel so dumb. This was so much easier than I thought it was.

**Reply by jkoutavas** [reply]
> You can run this code with RNW alongside existing React DOM code

Can the base app remain React, and have it use React Native components? Or, does it require a base app to be RNW and then  have React and React Native Web components co-exist?

**Reply by just-Bri** [reply]
> > You can run this code with RNW alongside existing React DOM code
> 
> Can the base app remain React, and have it use React Native components? Or, does it require a base app to be RNW and then have React and React Native Web components co-exist?

What I did was literally just install react-native and react-native-web in my current react app. The only config I had to change was adding react-native in my declaration.d.ts:  
declare module react-native;

And it rendered with correct styling etc.  
This could be an absolute game changer for my work flow moving forward! 


Sidenote: our we pack config already had a bunch of loaders in place for TS/babel etc. Not sure if any of those would be required for RNW but it basically worked out of the box for me.
